File: solutions/2957. Remove Adjacent Almost-Equal Characters/2957.py
class Solution:
  def removeAlmostEqualCharacters(self, word: str) -> int:
    ans = 0
    i = 1
    while i < len(word):
      if abs(ord(word[i]) - ord(word[i - 1])) <= 1:
        ans += 1
        i += 2
      else:
        i += 1
    return ans

File: solutions/2957. Remove Adjacent Almost-Equal Characters/2957.cpp
class Solution {
 public:
  int removeAlmostEqualCharacters(string word) {
    int ans = 0;

    int i = 1;
    while (i < word.length())
      if (abs(word[i] - word[i - 1]) <= 1) {
        ++ans;
        i += 2;
      } else {
        i += 1;
      }

    return ans;
  }
};
